devimint/src/tests.rs

```diff
@@ -1504,6 +1504,95 @@ pub async fn gw_reboot_test(dev_fed: DevFed, process_mgr: &ProcessManager) -> Re
     Ok(())
 }
 
+async fn gw_leave_fed_test_one_gw(fed: &Federation, gw: &Gatewayd, gw_name: &str) -> Result<()> {
+    let fed_id = fed.federation_id().await;
+
+    let gateway_balance = cmd!(gw, "balance", "--federation-id={fed_id}")
+        .out_json()
+        .await?
+        .as_u64()
+        .unwrap();
+
+    assert_eq!(
+        gateway_balance, 0u64,
+        "Balance should have been zero for {gw_name} gateway."
+    );
+
+    info!(
+        LOG_DEVIMINT,
+        "{gw_name} attempting to leave federation unforced while having a zero balance."
+    );
+    cmd!(gw, "leave-fed", "--federation-id={fed_id}")
+        .run()
+        .await?;
+    info!(LOG_DEVIMINT, "{gw_name} leave successful.");
+
+    info!("Reconnecting {gw_name} gateway to federation.");
+    gw.connect_fed(fed).await?;
+
+    info!("Adding payment to {gw_name} gateway.");
+    fed.pegin_gateway(99_999, gw).await?;
+
+    let gateway_balance = cmd!(gw, "balance", "--federation-id={fed_id}")
+        .out_json()
+        .await?
+        .as_u64()
+        .unwrap();
+
+    assert_ne!(
+        gateway_balance, 0u64,
+        "{gw_name} balance should have been non-zero."
+    );
+    info!(LOG_DEVIMINT, "{gw_name} balance is non-zero.");
+
+    info!(
+        LOG_DEVIMINT,
+        "{gw_name} attempting to leave federation unforced while having a non-zero balance."
+    );
+    cmd!(gw, "leave-fed", "--federation-id={fed_id}")
+        .run()
+        .await
+        .expect_err("{gw_name} should have errored, gateway is trying to leave with a non-zero balance without the --forced flag.");
+    info!(
+        LOG_DEVIMINT,
+        "{gw_name} unsuccessfully left federation unforced while having a non-zero balance."
```

Comment:
Nit: this statement is a bit confusing to me, I would expect "unsuccessfully" to indicate an error